dev/RadioButtons/InteractionTests/RadioButtonsTests.cs

```diff
@@ -42,132 +46,448 @@ public void TestCleanup()
             TestCleanupHelper.Cleanup();
         }
 
-        public List<string> m_colors = new List<string>
-        {
-            "Red",
-            "Orange",
-            "Yellow",
-            "Green",
-            "Blue",
-            "Indigo",
-            "Violet"
-        };
-
         [TestMethod]
         public void SelectionTest()
         {
             using (var setup = new TestSetupHelper("RadioButtons Tests"))
             {
-                Log.Comment("Testing initial selection");
-                VerifyColorSelected("Orange");
+                elements = new RadioButtonsTestPageElements();
+                SetItemType(RadioButtonsSourceType.RadioButton);
+                foreach (RadioButtonsSourceLocation location in Enum.GetValues(typeof(RadioButtonsSourceLocation)))
+                {
+                    SetSource(location);
 
-                Select("Red");
+                    SelectByIndex(1);
+                    VerifySelectedIndex(1);
+                    RadioButton item1 = FindElement.ByName<RadioButton>("Radio Button 1");
+                    Verify.IsTrue(item1.IsSelected);
 
-                VerifyColorSelected("Red");
-                
-                Log.Comment("Select item by changing SelectedItem");
-                Button selectItemBlue = FindElement.ByName<Button>("SelectItemBlue");
-                selectItemBlue.Click();
-                Wait.ForIdle();
+                    SelectByItem(3);
+                    VerifySelectedIndex(3);
+                    RadioButton item3 = FindElement.ByName<RadioButton>("Radio Button 3");
+                    Verify.IsTrue(item3.IsSelected);
+                    Verify.IsFalse(item1.IsSelected);
+                }
+            }
+        }
 
-                VerifyColorSelected("Blue");
+        //[TestMethod] Failing with a crash, todo: investigate.
```

Comment:
Is it intentional that the test is disabled? If there is a crash, perhaps leave a bug id in the comment.

Reply:
Yes, the test crashes currently but I wanted to get the review out. I'm going to debug a bit more but if/when I open an issue on the test I'll add the tag here.

---
In reply to: [347393083](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347393083) [](ancestors = 347393083)